codex/sales/slotqueue.nim

```diff
@@ -332,6 +357,17 @@ proc dispatch(self: SlotQueue,
       # throw because it is caller-defined
       warn "Unknown error processing slot in worker", error = e.msg
 
+proc clearSeenFlags*(self: SlotQueue) =
+  # To avoid issues with new queue items being pushed to the queue while all
+  # items are being iterated (eg if a new storage request comes in and pushes
+  # new slots to the queue), first pop all items in the queue off and
```

Comment:
Not finished comment ;-)

> first pop all items

Hmm I don't think this is what is happening bellow right? `self[i]` only access `i-th` element and not "pop" it... Right?

Reply:
Yep i figured out a better way and forgot to update the comment. Will update.